core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -300,6 +304,7 @@ public void onTopologyUpdate(final CacheTopology cacheTopology, final boolean is
       // Ensures writes to the data container use the right consistent hash
       // No need for a try/finally block, since it's just an assignment
       stateTransferLock.acquireExclusiveTopologyLock();
+      triangleOrderManager.updateCacheTopology(cacheTopology);
```

Comment:
You're right, having the ReceiverSequencer topology id updated only after it sees the first command with the new topology means SCI's cacheTopology was already updated and any commands with the old topology will throw an OutdatedTopologyException. Please add some comments to help me remember this the next time I read it :)